src/VscPreviewExtension/src/extension.ts

```diff
@@ -2,44 +2,93 @@
+    let tokenTreeProviderRegistration = workspace.registerTextDocumentContentProvider(tokenTreeScheme, tokenTreeProcess.provider);
 
     // Event register
     let showPreviewRegistration = commands.registerCommand("DFM.showPreview", uri => showPreview(dfmProcess));
     let showPreviewToSideRegistration = commands.registerCommand("DFM.showPreviewToSide", uri => showPreview(dfmProcess, uri, true));
     let showSourceRegistration = commands.registerCommand("DFM.showSource", showSource);
+    let showTokenTreeToSideRegistration = commands.registerCommand("DFM.showTokenTreeToSide", uri => showTokenTree(tokenTreeProcess));
 
-    context.subscriptions.push(showPreviewRegistration, showPreviewToSideRegistration, showSourceRegistration, providerRegistration);
+    context.subscriptions.push(showPreviewRegistration, showPreviewToSideRegistration, showSourceRegistration, showTokenTreeToSideRegistration);
+    context.subscriptions.push(previewProviderRegistration, tokenTreeProviderRegistration);
 
     workspace.onDidSaveTextDocument(document => {
         if (isMarkdownFile(document)) {
             const uri = getMarkdownUri(document.uri);
             dfmProcess.callDfm(uri);
+
+            tokenTreeProcess.callDfm(getTokenTreeUri(document.uri));
         }
     });
 
     workspace.onDidChangeTextDocument(event => {
         if (isMarkdownFile(event.document)) {
             const uri = getMarkdownUri(event.document.uri);
             dfmProcess.callDfm(uri);
+            // TODO: make token tree change synchronous
+            // tokenTreeProcess.callDfm(getTokenTreeUri(event.document.uri));
         }
     });
 
     workspace.onDidChangeConfiguration(() => {
         workspace.textDocuments.forEach(document => {
             if (document.uri.scheme === markdownScheme) {
                 dfmProcess.callDfm(document.uri);
+            } else if (document.uri.scheme === tokenTreeScheme) {
+                tokenTreeProcess.callDfm(document.uri);
             }
         });
     });
+
+    window.onDidChangeTextEditorSelection(event => {
+        startLine = event.selections[0].start.line + 1;
+        endLine = event.selections[0].end.line + 1;
+    });
+
+    // Http server to communicate with js
+    let http = require("http");
+    let server = http.createServer();
+    server.on("request", function (req, res) {
+        let requestInfo = req.url.split("/");
+        if (requestInfo[1] === "lineNumber") {
+            mapToSelection(parseInt(requestInfo[2]), parseInt(requestInfo[3]));
+            res.writeHead(200, { "Content-Type": "text/plain" });
+            res.end("success");
+        } else {
+            res.writeHead(200, { "Content-Type": "text/plain" });
+            res.write(startLine + " " + endLine);
+            res.end();
+        }
+    });
+    try {
+        server.listen(port);
+    }
+    catch (e) {
+        window.showErrorMessage(e);
+    }
+}
+
+function mapToSelection(startLineNumber, endLineNumber) {
```

Comment:
>startLineNumber [](start = 24, length = 15)

need check startLineNumber and endLineNumber? start shuold no larger than end, and no less than 1?